File: 프로그래머스/unrated/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.js
function solution(n) {
    const result = [n];
    while(n !== 1){
        n % 2 === 0 ? n /= 2 : n = 3 * n + 1;
        result.push(n)
    }
    return result
    
    // 재귀적 풀이
    // arr.push(n)
    // if (n === 1) return arr
    // if (n % 2 === 0) return solution(n / 2, arr)
    // return solution(3 * n + 1, arr)
}

File: 프로그래머스/unrated/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.py
def solution(n, arr = []): 
    # arr.append(n);
    # if n == 1:
    #     return arr;
    # if n % 2 == 0:
    #     return solution(n / 2, arr)
    # return solution(3 * n + 1, arr)
    
    result = [n]
    while n > 1:
        if n % 2 == 0:
            n /= 2
        else:
            n = 3 * n + 1
        result.append(n)
    return result